### DISCUSSION TITLE
numpy 1.22 support?

### DISCUSSION THREAD

**Discussion by fkiraly** [alan-turing-institute/sktime#2037] [Q&A]
from #1921, @ilkersigirci:

"Is there any specific reason not to include numpy version 1.22 ?"

**Comment by fkiraly** [comment]
@ilkersigirci,

numpy 1.22 is relatively new, and `numba`, one of our depdencies, does not support it yet:
https://github.com/numba/numba/issues/7754

Further, once dependencies start unlocking support for numpy 1.22, there is a good likelihood of things breaking around dependencies making numpy 1.22 compatible releases.

We try to be a bit defensive here based on earlier bad experience, and relax the upper bounds around releases, see also
https://github.com/alan-turing-institute/sktime/issues/1480

That is, we´ll test at every release and if all is fine we move the upper bound to (probably) 1.23.

**Comment by ilkersigirci** [comment]
@fkiraly Numba released [0.55.2](https://github.com/numba/numba/releases/tag/0.55.2) with numpy 1.22 support. So I guess sktime soon support it, too?

Note: In my local environment, I have successfully installed sktime with numba==0.55.2 and numpy==1.22.4. FYI
